components/org.wso2.carbon.identity.conditional.auth.functions.user/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/user/UpdateUserPasswordFunctionImpl.java

```diff
@@ -57,29 +58,46 @@ public void updateUserPassword(JsAuthenticatedUser user, Object... parameters) {
 
         char [] newPassword = null;
         Map<String, Object> eventHandlers = null;
+        boolean skipPasswordValidation = false;
 
-        if (parameters.length == 2) {
-            LOG.debug("Both password and event handlers are provided.");
+        if (parameters.length == 1) {
+            LOG.debug("Only the password is provided.");
             newPassword = ((String) parameters[0]).toCharArray();
-
-            if (parameters[1] instanceof Map) {
-                eventHandlers = (Map<String, Object>) parameters[1];
-            } else {
-                throw new IllegalArgumentException("Invalid argument type. Expected eventHandlers " +
-                        "(Map<String, Object>).");
-            }
         } else {
-            LOG.debug("Only the password is provided.");
+            LOG.debug("Password is provided.");
             newPassword = ((String) parameters[0]).toCharArray();
+
+            for (int i = 1; i < parameters.length; i++) {
+                Object parameter = parameters[i];
+                if (parameter instanceof Map) {
+                    LOG.debug("Event handlers are provided.");
+                    eventHandlers = (Map<String, Object>) parameter;
+                } else if (parameter instanceof Boolean) {
+                    LOG.debug("SkipPasswordValidation flag are provided.");
+                    skipPasswordValidation = (Boolean) parameter;
+                } else {
+                    throw new IllegalArgumentException(
+                            "Invalid argument type. Expected eventHandlers (Map<String, Object>) or skipPasswordValidation(Boolean)."
+                    );
+                }
+            }
```

Comment:
When there are multiple parameters, check the standard way of resolving the params (In other adaptive functions). 
Whether the params are defined with the order they are passed to the function.

If not iterating a for loop is not good as we can pass lengthy boolean array to the adaptive script and it might loop through which can cause impact for the server.